app/adapters/osu_beatmap_file.py

```
@@ -0,0 +1,508 @@
+            continue
+
+        # pre-v6 files omit `uninherited`; a positive beatLength implies it.
+        uninherited = True
+        if len(fields) >= 7:
+            uninherited = fields[6].strip() not in ("0", "")
+
+        if uninherited and beat_length > 0:
+            points.append((time, beat_length))
+
+    if not points:
+        raise OsuFileParseError("no uninherited timing points")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)